Fix test `git_get_commit`

progician opened the pull request:
Changelog: omit
Docs: omit

Fixes #13104 

- [x] Refer to the issue that supports this Pull Request.
- [ ] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [ ] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. 

<sup>**Note:** By default this PR will skip the slower tests and will use a limited set of python versions. Check [here](https://github.com/conan-io/conan/blob/develop/.github/PR_INCREASE_TESTING.md) how to increase the testing level by writing some tags in the current PR body text.</sup>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/conan-io/conan?pullRequest=13105) <br/>All committers have signed the CLA.

progician committed:
test: Add xfail on the failing test

The test is only failing in the case of the default branch on the system
running the test is other than "master". To simulate this, you must
configure your global git to have, say, "main" as the default branch:

    git config --global init.defaultBranch main

After this, you can simply run this test case:

    python -m tox -- python -m tox -- conans/test/unittests/tools/scm/test_git_get_commit.py

The xfail will make it an expected failure.

progician committed:
fix(test): Create "master" branch after repo init

When initialising a new repository, Git will consider the
init.defaultBranch configuration setting that can be something other
than "master".

The git init command however accepts an option called "--init-branch"
which we can use to set the default branch explicitly. That solution,
however, can only work with newer Git versions.

The more versatile solution is to create the master branch if it does
not exist.

This way, whatever is the setting on the developer machine, the test
will pass.

Another solution could be to get the value of the configuration like so:

    git config --get init.defaultBranch

And use the output of the command above to checkout the default branch.
However, it would take a more code to pull this one out so I've decided
that the explicit setting of the default branch in the init call is the
most economical solution.

progician commented:
The linux failure seems to stem from the fact that the Conan CI uses a Linux machine/image that has an older Git version than the other platforms. Probably it would be a good idea to match all 3 platforms, to expect tests to have the same environment.

What would you recommend me to do here? The problem is that this issue comes about after Git 2.28 which introduced the possibility to change the default branch: See here in the [feature highlights](https://github.blog/2020-07-27-highlights-from-git-2-28/#introducing-init-defaultbranch)

memsharded commented:
Thanks for your contribution. Yes, it would be nice to fix this.

The thing is that it is not that bad to have different git versions, actually it helps to identify this kind of issues, that otherwise would hit users using older or newer versions, and we would not be aware.

> What would you recommend me to do here? The problem is that this issue comes about after Git 2.28 which introduced the possibility to change the default branch: See here in the [feature highlights](https://github.blog/2020-07-27-highlights-from-git-2-28/#introducing-init-defaultbranch)

Maybe something like:
```
git init .
git checkout -b master
```
would work in both modern and older git versions?

memsharded commented:
Thanks for fixing this!
